WeatherApp/Controller/LocationWeatherViewController.swift

```diff
@@ -0,0 +1,172 @@
+//
+//  LocationWeatherViewController.swift
+//  WeatherApp
+//
+//  Created by Marcin Wójciak on 17/01/2020.
+//  Copyright © 2020 Marcin Wójciak. All rights reserved.
+//
+
+import UIKit
+import CoreLocation
+
+class LocationWeatherViewController: UITableViewController, CLLocationManagerDelegate, WeatherManagerDelegate {
+    
+    let userDefaults = UserDefaults.standard
+    let locationManager = CLLocationManager()
+    var weatherManager = WeatherManager()
+    
+    var userLocations = [String]()
+    var weatherData = [WeatherModel]()
+    
+    override func viewWillAppear(_ animated: Bool) {
+        
+        
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        tableView.register(UINib(nibName: "LocationWeatherCell", bundle: nil), forCellReuseIdentifier: "LocationWeatherCell")
+        
+        weatherManager.delegate = self
+        
+        locationManager.requestWhenInUseAuthorization()
+        locationManager.desiredAccuracy = kCLLocationAccuracyThreeKilometers
+        locationManager.delegate = self
+        locationManager.requestLocation()
+        
+        if userLocations.count == 0 {
+            userLocations = userDefaults.stringArray(forKey: "UserLocations") ?? []
+        }
+        print(userLocations)
+        
+        for location in userLocations {
+            weatherManager.fetchWeatherData(for: location)
+        }
+        
+        // Uncomment the following line to preserve selection between presentations
+        // self.clearsSelectionOnViewWillAppear = false
```

Comment:
Komentarzy unikaj, bo czasem zmniejszają czytelność kodu, bo kod się często zmienia, a o aktualizacji komentarzy mało kto pamięta :P Przydaję się za to, gdy zależy nam na optymalizacji kodu i przez to kod staje się skomplikowany. W takich przypadkach przydaje się wyjaśnienie zastosowanego algorytmu.